services/core/events/blockchain.js

```diff
@@ -15,26 +15,29 @@
  */
 const logger = require('lisk-service-framework').Logger();
 const { SocketClient } = require('lisk-service-framework');
+const Signal = require('signals');
 
 const core = require('../shared/core');
-const recentBlocksCache = require('../shared/recentBlocksCache');
-const delegateCache = require('../shared/delegateCache');
-
 const config = require('../config');
 
 const coreSocket = SocketClient(config.endpoints.liskWs);
 logger.info(`Registering ${config.endpoints.liskWs}`);
 
-recentBlocksCache.init(core);
+const signals = {
+	blockCached: new Signal(),
+};
 
 module.exports = [
 	{
 		name: 'block.change',
 		description: 'Keep the block list up-to-date',
 		controller: callback => {
 			coreSocket.socket.on('blocks/change', async data => {
-				logger.debug('Returning block list to the socket.io client...');
+				logger.debug(`New block arrived (${data.id})...`);
 				const restData = await core.getBlocks({ blockId: data.id });
+				core.setLastBlock(restData.data[0]);
+				signals.blockCached.dispatch(restData.data[0]);
+				core.reloadBlocks(202);
```

Comment:
Can we refer to the `config.cacheNumOfBlocks` property instead of the hardcoded value?

Reply:
Sure, I'll fix it soon.